src/Game/GameObjects/PlayerMobile.cs

```diff
@@ -187,6 +199,53 @@ private Item FindItemInContainerRecursive(Item container, ushort graphic)
             return found;
         }
 
+        private Item FindItemByClilocInContainerRecursive(Item container, int cliloc)
+        {
+            Item found = null;
+
+            if (container != null)
+            {
+                for (LinkedObject i = container.Items; i != null; i = i.Next)
+                {
+                    Item item = (Item) i;
+
+                    if (item.NameCliloc == cliloc)
+                    {
+                        return item;
+                    }
+
+                    if (!item.IsEmpty)
+                    {
+                        found = FindItemByClilocInContainerRecursive(item, cliloc);
+
+                        if (found != null && found.NameCliloc == cliloc)
+                        {
+                            return found;
+                        }
+                    }
+                }
+            }
+
+            return found;
+        }
+
+        public Item FindPreferredItemByCliloc(params int[] clilocs)
```

Comment:
`params int[]` creates an array internally every time you call the function.
So I sugest to replace it with `Span<int>`.
The function will be called using `Span<int> span = stackalloc int[] { 123, 456, 789 }`

Reply:
Memory management isn't something I've dealt with so thanks for the suggestion. I just implemented this in the new commit https://github.com/ClassicUO/ClassicUO/pull/1534/commits/036166676ff573b7b743ea463bf255a457a41382 but I'm not sure about it.

I can't do `Span<int> clilocs = stackalloc int[] { 1, 2, 3 };` for each macro since `clilocs` is already defined after the first time. Using `clilocs1`, and then `clilocs2`, feels bad but I guess it would work.

I tried this pattern
```csharp
case MacroType.UseObject:
    Item obj;
    Span<int> clilocs;

    switch (macro.SubCode)
    {
        case MacroSubType.BestHealPotion:
            clilocs = stackalloc int[3];
```
Which gives me "A result of a stackalloc expression of type 'Span<int>' cannot be used in this context because it may be exposed outside of the containing method"

So I settled on what I have, hope it's OK.

Reply:
It’s fine allocating fixed local array.
Or you can also try to put the body of eaxh case into a new scope using: { 
// code
}

Reply:
Yeah I was overthinking it with the local definitions, it's nicer now. All ready I think, commits squashed